orangecontrib/text/tests/test_twitter.py

```diff
@@ -0,0 +1,216 @@
+
+        def on_start():
+            self.assertEqual(self.checker, 0)
+            self.checker += 1
+
+        def on_progress(progress):
+            self.assertEqual(self.checker, progress)
+            self.checker += 1
+
+        def on_finish():
+            self.assertEqual(self.checker, 3)
+            self.checker += 1
+
+        api = twitter.TwitterAPI(self.credentials, on_start=on_start,
+                                 on_progress=on_progress, on_finish=on_finish)
+        api.search(word_list=['hello'], max_tweets=2, lang='en')
+        api.join()
+        self.assertEqual(self.checker, 4)
+
+    def test_create_corpus(self):
+        self.api.search(word_list=['hello'], max_tweets=5)
+        self.api.join()
+        corpus = self.api.create_corpus()
+        self.assertIsInstance(corpus, Corpus)
+        self.assertEqual(len(corpus), 5)
+
+    def test_crate_corpus_attr_selection(self):
+        self.api.search(word_list=['hello'], max_tweets=5)
+        self.api.join()
+        attributes = ['text', 'created_at', 'author_id']
+        corpus = self.api.create_corpus(included_attributes=attributes)
+        domain_attributes = [attr.name for attr in chain(corpus.domain.attributes, corpus.domain.metas)]
+        self.assertEqual(len(domain_attributes), 3)
+        for attr in attributes:
+            self.assertIn(attr, domain_attributes)
+
+    def test_clear(self):
+        self.api.search(word_list=['hello'], max_tweets=5)
+        self.api.join()
+        self.assertEqual(len(self.api.container), 5)
+
+        self.api.reset()
+        self.assertEqual(len(self.api.container), 0)
+
+    def test_report(self):
+        self.api.search(word_list=['hello'], max_tweets=5)
+        self.api.join()
+        self.assertEqual(len(self.api.history), 1)
+        self.assertIsNotNone(self.api.task.report())
+
+    def test_geo_util(self):
+        point = twitter.coordinates_geoJSON({})
+        self.assertIsNone(point[0])
+        self.assertIsNone(point[1])
+
+        point = twitter.coordinates_geoJSON({'coordinates': [10, 10]})
+        self.assertEqual(point[0], 10)
+        self.assertEqual(point[1], 10)
+
+    def test_build_query(self):
+        # https://dev.twitter.com/rest/public/search
+
+        query = self.api.build_query(word_list=['hello', 'world'])
+        self.assertIn('hello', query)
+        self.assertIn('world', query)
+
+        query = self.api.build_query(authors=['johndoe'])
+        self.assertIn('from:johndoe', query)
+
+        query = self.api.build_query(since=date(2016, 10, 9))
+        self.assertIn('since:2016-10-09', query)
+
+        query = self.api.build_query(until=date(2016, 10, 9))
+        self.assertIn('until:2016-10-09', query)
+
+
+@unittest.skipIf(NO_CREDENTIALS, CREDENTIALS_MSG)
+class TestSearch(unittest.TestCase):
+    def setUp(self):
+        self.credentials = get_credentials()
+        self.api = twitter.TwitterAPI(self.credentials)
+
+    def test_contains(self):
```

Comment:
I still don't completely understand why this test (and also many below) could not be run on Travis without credentials. If `tweepy.Cursor` is mocked and we provide sufficient data in cache then `TwitterAPI.search` method should return the desired data and test should work. Missing credentials shouldn't be a problem since we are not accessing Twitter anyway.

I haven't tested with Travis but locally, without credentials, only 2 out of 7 tests fail and even those two could probably be fixed. Even if not, we should probably run the remaining 5 on Travis.

@s-alexey if you still think we shouldn't run this on Travis, can you try to explain again why not?